main.py

```diff
@@ -67,40 +67,42 @@ def main():
                 for sid in ROSTERS[course_id]:
                     print_information(f"- {sid}: {STUDENTS[sid]}")
 
+                
                 #remove the cast to an int, to check if its an empty string
                 student_id = input("Enter Student ID: ")
                 while(student_id == ""):
                     print("You must enter a student id! ")
                     student_id = input("Enter Student ID: ")
 
-                #cast the string back into an int
-                student_id = int (student_id)
-
-
-                isGradeEmpty = True
-                while (isGradeEmpty):
-                    grade = input("Enter Grade: ") 
-
-                    if (not grade or grade == "" or grade.startswith(" ")):
-                        print("\tGrade cannot be empty")
-                        continue
-                        
-                    else: 
-                        isGradeEmpty = False
-
+                #trys to cast the string back into an int
                 try:
-                    grade_value = float(grade)
-                    if grade_value < 0:
-                        print_error("Grade cannot be negative.")
+                    student_id = int (student_id)
+                except ValueError:
+                        print_error("Invalid ID format. Please enter a number.")
+                        continue  # Go back to menu
+                
+                if student_id in ROSTERS[course_id]:
+                    isGradeEmpty = True
+                    while (isGradeEmpty):
+                        grade = input("Enter Grade: ") 
+                        if (not grade or grade == "" or grade.startswith(" ")):
+                            print("\tGrade cannot be empty")
+                            continue
+                                
+                        else: 
+                            isGradeEmpty = False
+
+                    try:
+                        grade_value = float(grade)
```

Comment:
Since you only allow float type of grades, please update all the tests to make sure all the grades in the tests are float type. Otherwise, I would be confused by those tests and thought the grades can be letter grades. Thanks!